### ISSUE TITLE
Broken text rendering

### ISSUE THREAD

**Issue by vkcz** [skyjake/lagrange#311]
Lagrange sometimes fails at rendering text (missing characters and lines out of place; see images) in the main text display, side interfaces, and context menus.
This is somehow dependent on text scaling: text rendering frequently breaks down or fixes itself when zoom-level is changed.

Things I have tried that don't fix it:
* changing the font
* enabling "monospace body"
* changing line width
* disabling "wrap plain text"

This happens on Lagrange 1.5.2, installed on Windows 7 from `lagrange_v1.5.2_windows-x64_setup.exe`.

![image](https://user-images.githubusercontent.com/69770865/125717989-fbf45588-0a99-45c3-a0cc-99a774c61c80.png)
![image](https://user-images.githubusercontent.com/69770865/125718025-0e6201de-ad7f-42dc-bbe5-f44d71c73515.png)

**Comment by skyjake** [comment]
This is a bug in the text renderer’s font glyph cache. I have seen it occasionally myself, but perhaps because I tend to restart the app frequently, It’s happening rarely for me.

This does need fixing. I’ll see if it can be reproduced more easily by reducing the cache size.

**Comment by skyjake** [comment]
So far I've been unable to reproduce this issue. Are there any usage patterns that you can identify where the issue would start occurring?

**Comment by vkcz** [comment]
Starting Lagrange again (after a while of inactivity), there were no rendering issues at all, until I changed the zoom/scaling by a few scroll-units. After that, some further rescaling sometimes returned rendering to correctness, and sometimes back to chaos.
tl;dr Issue is triggered for me by changing page scale

**Comment by skyjake** [comment]
Thanks, that should help with debugging this!

**Comment by skyjake** [comment]
v1.6 makes a number of changes in text rendering so this issue may already be fixed. However, I haven't found a way to reliably reproduce the problem.

**Comment by skyjake** [comment]
I am assuming the issue has been fixed in v1.6 after the introduction of HarfBuzz and other text rendering changes.
